arm-containerregistry/2016-06-27-preview/swagger/containerregistry.json

```diff
@@ -0,0 +1,642 @@
+          "readOnly": true
+        },
+        "creationDate": {
+          "format": "date-time",
+          "description": "The creation date of the container registry in ISO8601 format.",
+          "type": "string",
+          "readOnly": true
+        },
+        "adminUserEnabled": {
+          "description": "The boolean value that indicates whether admin user is enabled.",
+          "type": "boolean",
+          "readOnly": true
+        },
+        "storageAccount": {
+          "$ref": "#/definitions/StorageAccountBase",
+          "description": "The storage account base properties.",
+          "readOnly": true
+        }
+      }
+    },
+    "StorageAccountBase": {
+      "description": "The storage account base properties.",
+      "properties": {
+        "name": {
+          "description": "Storage account name.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "RegistryCreateParameters": {
+      "description": "The parameters for creating a container registry.",
+      "required": [
+        "location",
+        "properties"
+      ],
+      "properties": {
+        "location": {
+          "description": "Resource location.",
+          "type": "string"
+        },
+        "tags": {
+          "description": "Resource tags.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
+          "$ref": "#/definitions/RegistryPropertiesCreateParameters",
+          "description": "The parameters for creating container registry properties.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "RegistryPropertiesCreateParameters": {
+      "description": "The parameters for creating container registry properties.",
+      "required": [
+        "tenantId",
+        "storageAccount"
+      ],
+      "properties": {
+        "tenantId": {
+          "description": "The tenant id for the current user.",
+          "type": "string"
+        },
+        "adminUserEnabled": {
+          "description": "The boolean value that indicates whether admin user is enabled.",
+          "type": "boolean"
+        },
+        "storageAccount": {
+          "$ref": "#/definitions/StorageAccountProperties",
+          "description": "The storage account properties."
+        }
+      }
+    },
+    "StorageAccountProperties": {
+      "description": "The storage account properties.",
+      "required": [
+        "name",
+        "accessKey",
+        "endPointUrl"
+      ],
+      "properties": {
+        "name": {
+          "description": "Storage account name.",
+          "type": "string"
+        },
+        "accessKey": {
+          "description": "Storage account access key.",
+          "type": "string"
+        },
+        "endPointUrl": {
+          "description": "Storage account endpoint URL.",
+          "type": "string"
+        },
+        "location": {

```

Comment:
One of our requirements is that the registry has to be in the same location as the storage account. So either the request will have location or the RP will query storage account location using storage account name (also the reason why we want to have storage account name as a property for registry). We probably want to remove it in the future, since there is no guarantee the provided location is the actual location of the storage account, so we made it optional here.